src/components/application_manager/src/policies/policy_handler.cc

```diff
@@ -1493,12 +1493,31 @@ void PolicyHandler::OnSnapshotCreated(const BinaryMessage& pt_string) {
     return;
   }
 
-  AppIdURL app_url = policy_manager_->GetNextUpdateUrl(urls);
-  while (!IsUrlAppIdValid(app_url.first, urls)) {
-    app_url = policy_manager_->GetNextUpdateUrl(urls);
+  const uint32_t app_id = GetAppIdForSending();
+  const std::string app_policy_id =
+      application_manager_.application(app_id)->policy_app_id();
+
+  size_t app_idx = 0;
+  for (; app_idx < urls.size(); ++app_idx) {
+    if (helpers::Compare<std::string, helpers::EQ, helpers::ONE>(
+            policy::kDefaulgit tId,
+            urls[app_idx].app_id,
+            urls[app_idx].app_id,
+            app_policy_id) &&
+        IsUrlAppIdValid(app_idx, urls))
+      break;
+  }
+
+  if (app_idx >= urls.size()) {
+    LOG4CXX_ERROR(logger_, "All applications unavailable!");
+    return;
+  }
+
+  for (size_t url_idx = 0; url_idx < urls[app_idx].url.size(); ++url_idx) {
```

Comment:
@ZhdanovP  you can use more compact range based loop
for (size_t url_idx : urls[app_idx].url.size())